### PR TITLE
QE Coverage : Accesscontrol - CRD Roles

### PR DESCRIPTION
**Pull request by bnshr**
https://issues.redhat.com/browse/CNFCERT-622

* Implement negative tests for CRD roles
* Refactoring of role and custom resource creation

### CODE REVIEW

File: tests/accesscontrol/tests/access_control_pod_role_bindings.go [hunk 1]
```diff
@@ -16,7 +16,7 @@ func setupInitialRbacConfiguration(namespace string) {
 	err := globalhelper.CreateServiceAccount(tsparams.TestServiceAccount, namespace)
 	Expect(err).ToNot(HaveOccurred())
 
-	err = globalhelper.CreateRole(tsparams.TestRoleName, namespace)
+	err = globalhelper.CreateRole(globalhelper.DefineRole(tsparams.TestRoleName, tsparams.TestAccessControlNameSpace))
```
Comment (sebrandon1) [inline tests/accesscontrol/tests/access_control_pod_role_bindings.go:19]: This seems to be a mistake/leftover.

**Review by sebrandon1 — COMMENTED** [review]
One comment about an accesscontrol change

File: tests/accesscontrol/tests/access_control_pod_role_bindings.go (hunk 1 above)
Comment (bnshr) [reply tests/accesscontrol/tests/access_control_pod_role_bindings.go]: Changed to the namespace arg.

File: tests/accesscontrol/tests/access_control_crd_roles.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+		err = globalhelper.LaunchTests(tsparams.TnfCrdRoles,
+			globalhelper.ConvertSpecNameToFileName(CurrentSpecReport().FullText()))
+		Expect(err).To(HaveOccurred())
+
+		By("Verify test case status in Junit and Claim reports")
+		err = globalhelper.ValidateIfReportsAreValid(tsparams.TnfCrdRoles, globalparameters.TestCaseFailed)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("Custom resource is deployed, with improper role [negative test]", func() {
```
Comment (theemj) [inline tests/accesscontrol/tests/access_control_crd_roles.go:125]: For skip cases the convention is to end the testcase name with [skip], as in:
```suggestion
	It("Custom resource is deployed, with improper role [skip]", func() {
```

File: tests/accesscontrol/tests/access_control_crd_roles.go [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+		err = globalhelper.LaunchTests(tsparams.TnfCrdRoles,
+			globalhelper.ConvertSpecNameToFileName(CurrentSpecReport().FullText()))
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Verify test case status in Junit and Claim reports")
+		err = globalhelper.ValidateIfReportsAreValid(tsparams.TnfCrdRoles, globalparameters.TestCasePassed)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("Custom resource is deployed, one role defined with multiple api groups [negative test]", func() {
```
Comment (theemj) [inline tests/accesscontrol/tests/access_control_crd_roles.go:79]: to match the convention of the rest if the project these should just be "[negative]"

```suggestion
	It("Custom resource is deployed, one role defined with multiple api groups [negative]", func() {
```

File: tests/accesscontrol/tests/access_control_crd_roles.go [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+		err = globalhelper.LaunchTests(tsparams.TnfCrdRoles,
+			globalhelper.ConvertSpecNameToFileName(CurrentSpecReport().FullText()))
+		Expect(err).To(HaveOccurred())
+
+		By("Verify test case status in Junit and Claim reports")
+		err = globalhelper.ValidateIfReportsAreValid(tsparams.TnfCrdRoles, globalparameters.TestCaseFailed)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("Custom resource is deployed, one role with multiple resources defined [negative test]", func() {
```
Comment (theemj) [inline tests/accesscontrol/tests/access_control_crd_roles.go:102]:
```suggestion
	It("Custom resource is deployed, one role with multiple resources defined [negative]", func() {
```

**Review by theemj — APPROVED** [review]
couple small testcase title suggestions but otherwise looks good and works well :+1:

**Pull request merged**